provider/src/main/kotlin/org/gradle/kotlin/dsl/accessors/DefaultAccessorsProvider.kt

```diff
@@ -0,0 +1,28 @@
+/*
+ * Copyright 2018 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gradle.kotlin.dsl.accessors
+
+import org.gradle.kotlin.dsl.provider.spi.AccessorsProvider
+import org.gradle.kotlin.dsl.provider.spi.ProjectSchema
+import org.gradle.kotlin.dsl.provider.spi.TypeAccessibility
+
+
+class DefaultAccessorsProvider : AccessorsProvider {
+
+    override fun forEachAccessorOf(accessibleSchema: ProjectSchema<TypeAccessibility>, action: (String) -> Unit) =
+        accessibleSchema.forEachAccessor(action)
```

Comment:
This seems like a somewhat excessive amount of indirection.

Reply:
Good catch! `:provider-plugins` could depend on `:provider` directly.
I'll open a simplification PR shortly.